doc/ReferenceTables.md

```diff
@@ -0,0 +1,153 @@
+  UID are hardcoded to start at 0. But since the reserved UIDs are documented in
+  the `ReferenceKeyspace` entries, it's easy.
+
+* At this point, the destination shards will also replicate from the reference
+  keyspace. When the `vtctl MigrateServedType master` command is issued, it
+  needs to just remove the horizontal resharding Filtered Replication entries,
+  not the `ReferenceKeyspace` entries entries.
+
+## Other Use Cases
+
+Other scenarios might also need to be supported, or explicitely disabled:
+
+* Simple schema changes, or complicated Schema Swap in the reference keyspace:
+  They would also need to be applied to the destination keyspace, the same way.
+
+* Vertical Split of the reference keyspace: Since it is replicated, splitting it
+  will be more complicated.
+
+## Query Routing
+
+This would be handled by the vtgate and the VSchema. Once the reference tables
+are documented in the VSchema, vtgate will know to do the following:
+
+* DMLs on the reference tables are routed to the reference keyspace.
+
+* Select queries on the reference tables only are also routed to the reference
+  keyspace.
+  
+* JOIN queries between reference tables and destination keyspace tables can be
+  routed only to the right destination keyspace (based on that keyspace sharding
+  situation).
+
+Note this introduces some corner cases: for instance, if the client is asking
+for a JOIN between reference tables and destination keyspace tables, with tablet
+type `master`. Routing this to the destination keyspace would satisfy the
+critical read for the destination tables, but not for the reference
+tables. vtgate may need to perform the JOIN to both masters at this point.
+
+Action Items:
+
+* Find the right way to represent reference tables in the VSchema.
+
+* Implement corresponding query routing.
+
+## Notes
+
+### Vitess Keyspace vs MySQL Database
+
+This may force us to revisit the use of databases in our tablets. The current
+assumption is that a keyspace only has one MySQL database (with a name usually
+derived from the keyspace name with a `vt_` prefix, but that can also be
+changed):
+
+* When vttablet connects to MySQL for data queries, it uses that database name
+  by default.
+
+* The VSchema also maps tables to keyspaces, so it can just send queries that
+  have no keyspace to the right shard (which in turns is configured properly for
+  that database).
+
+* Vitess' Filtered Replication only replicates data related to that single
+  database. The database name has to be the same when we horizontally split a
+  keyspace, so statements from the source shards can be applied on the
+  destination shards.
+
+* Vitess' Query Service only loads the schema for that single database.
+
+Maybe it's time to change this assumption:
+
+* A keyspace could be defined as a group of databases, each having a group of
+  tables.
+
+* When addressing a table, we could support the `keyspace.database.table`
+  syntax.
```

Comment:
Yeah I was just putting that out there so we could think about it. I kinda think we need to decide on our view of the world, and stick to it. I think I'd be fine too with keyspace == database (the implicit way we have it now).

Or maybe we use v3 in a slightly different way, and have a keyspace be a container of tables, and create databases as groups of tables, that can be in any keyspace we want.

Either way, I think we need to decide and document / implement it.